src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -934,17 +991,33 @@ private void UpdateHyperlinks()
 			{
 				InsertHandler(PointerPressedEvent, OnPointerPressed);
 				InsertHandler(PointerReleasedEvent, OnPointerReleased);
+				InsertHandler(PointerMovedEvent, OnPointerMoved);
+				InsertHandler(PointerEnteredEvent, OnPointerEntered);
+				InsertHandler(PointerExitedEvent, OnPointerExit);
 				InsertHandler(PointerCaptureLostEvent, OnPointerCaptureLost);
 			}
 			else if (!HasHyperlink && previousHasHyperlinks)
 			{
 				RemoveHandler(PointerPressedEvent, OnPointerPressed);
 				RemoveHandler(PointerReleasedEvent, OnPointerReleased);
+				RemoveHandler(PointerMovedEvent, OnPointerMoved);
+				RemoveHandler(PointerEnteredEvent, OnPointerEntered);
+				RemoveHandler(PointerExitedEvent, OnPointerExit);
 				RemoveHandler(PointerCaptureLostEvent, OnPointerCaptureLost);
 			}
 		}
 
-		private bool HasHyperlink => _hyperlinks.Any();
+		private bool HasHyperlink
+		{
+			get
+			{
+				var hasHyperlink = _hyperlinks.Any();
+
+				Debug.Assert(!(!hasHyperlink && _hyperlinkOver is { }));
```

Comment:
Does this assert hold if you had a hyperlink, on on pointer over you removed it from the TextBlock Inlines?

If the assert wouldn't hold in this case, I suggest deleting it completely.

Reply:
Yes, because removing the hyperlink triggers `InlineCollection.OnCollectionChanged` which triggers `TextBlock.InvalidateInlines` which triggers `TextBlock.UpdateHyperlinks` which clears the removes all removed hyperlinks and their state.